metals/src/main/scala/scala/meta/internal/metals/Trace.scala

```diff
@@ -0,0 +1,107 @@
+package scala.meta.internal.metals
+
+import java.io.PrintWriter
+import java.nio.file.Files
+import java.nio.file.StandardOpenOption
+
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+import scala.meta.internal.io.PathIO
+import scala.meta.io.AbsolutePath
+
+import dev.dirs.ProjectDirectories
+
+/**
+ * Manages JSON-RPC tracing of incoming/outgoing messages via BSP and LSP.
+ */
+object Trace {
+
+  /**
+   * Returns a printer to trace JSON messages if the user opts into it.
+   */
+  def setup(protocolName: String): Option[PrintWriter] = {
+    MetalsLogger.redirectSystemOut(globalLog)
+    setupTracePrinter(protocolName)
+  }
+
+  def globalLog: AbsolutePath =
+    globalDirectoryOpt.getOrElse(localDirectory).resolve("global.log")
```

Comment:
I wonder if `globalLog` is necessary, it doesn't contain any useful information (only a few logs from the startup)

Reply:
I think we could remove it, I never used it